src/fclaw2d_convenience.c

```diff
@@ -276,6 +276,17 @@ fclaw2d_domain_new (p4est_wrap_t * wrap, sc_keyvalue_t * attributes)
     domain->local_minlevel = local_minlevel;
     domain->local_maxlevel = local_maxlevel;
 
+    /* determine maximum number of local patches */
+    domain->local_max_patches = 0;
+    for (i = 0; i < wrap->p4est->mpisize; ++i)
+    {
+        domain->local_max_patches = SC_MAX (domain->local_max_patches,
```

Comment:
This should work! I'm just wondering whether we should explicitly cast the difference of two p4est_gloidx_t to int and store the result in a temporary int variable to avoid double evaluation.

Reply:
I added a temporary variable for the difference, which is then used for an explicitly cast.